lib/parseTriggers.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var FirebaseError = require('./error');
+var fork = require('child_process').fork;
+var path = require('path');
+var RSVP = require('rsvp');
+
+var TRIGGER_PARSER = path.resolve(__dirname, './triggerParser.js');
+
+module.exports = function(projectId, dbInstance, sourceDir) {
+  return new RSVP.Promise(function(resolve, reject) {
+    var env = {
```

Comment:
FIREBASE_PROJECT is populated for "firebase serve" not "firebase deploy". Firebase serve needs it for both trigger parsing, and for running the function if `functions.config().firebase` is called. 

Firebase deploy does not need it yet, since it has the config file available. Later on, we will need to populate the GCF runtime with FIREBASE_PROJECT.

I also didn't use the new endpoint specified in the bug, since I want to de-risk the next release. I'm using the same endpoints that the CLI was using before to get the Firebase config info.